src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2439,6 +2439,14 @@ class SetStatementSegment(BaseSegment):
                     OneOf(
                         "ON",
                         "OFF",
+                        "LOW",
+                        "NORMAL",
+                        "HIGH",
+                        Ref("ParameterNameSegment"),
+                        OneOf(
+                            Ref("NumericLiteralSegment"),
+                            Ref("QualifiedNumericLiteralSegment"),
+                        ),
```

Comment:
Is there some less janky way to support both `10` and `-10`?

Even `SELECT -1;` in this dialect throws a parsing error.